sdk/playwrighttesting/microsoft-playwright-testing/src/utils/utils.ts

```diff
@@ -98,6 +99,20 @@ export const validateMptPAT = (
     if (result!.accountId !== claims!.aid) {
       validationFailureCallback(ServiceErrorMessageConstants.WORKSPACE_MISMATCH_ERROR);
     }
+    const currentTime = Date.now();
+    const expirationTime = claims.exp! * 1000;
+    const sevenDaysInMs = 7 * 24 * 60 * 60 * 1000;
+
+    if (
+      expirationTime - currentTime <= sevenDaysInMs &&
+      process.env[oneTimeEnviromentVariable.WARNING_MPT_PAT_CLOSE_TO_EXPIRY] !== "true"
+    ) {
+      const daysToExpiration = Math.floor((expirationTime - currentTime) / (24 * 60 * 60 * 1000));
+      const expirationDate = new Date(expirationTime).toLocaleDateString();
+      const expirationWarning = `Warning: The access token used for this test run will expire in ${daysToExpiration} days on ${expirationDate}. To avoid failures, generate a new token from the portal and update it. We recommend switching to Microsoft Entra ID for authentication, as it eliminates the need to manage access tokens or worry about expiration, simplifying your process and enhancing security.`;
+      console.warn(expirationWarning);
+      process.env[oneTimeEnviromentVariable.WARNING_MPT_PAT_CLOSE_TO_EXPIRY] = "true";
```

Comment:
I believe we call validateToken from globalSetup , we could identify the call and choose to only show this warning in global setup flow. This way we could avoid this env variable logic https://github.com/Azure/azure-sdk-for-js/blob/83efc34ecdef728a726bcf83014d4bfa1d665f4e/sdk/playwrighttesting/microsoft-playwright-testing/src/core/playwrightService.ts#L116

Reply:
I tried doing this but it gets printed multiple times.

Reply:
globalSetup is not initialized for mpt pat token

Reply:
globalSetup is not initialized for mpt pat token

Reply:
But we are calling ValidatePAT function from getServiceConfig function which can be distinguished from the other scenario which we call from reporting ? getServiceConfig is one time call ?

Reply:
i think every time a test is run using service config file, getServiceConfig  method is called.